cmd/snap/cmd_get.go

```diff
@@ -139,6 +139,76 @@ func flattenConfig(cfg map[string]interface{}, root bool) (values []ConfigValue)
 	return values
 }
 
+func (c *cmdGet) outputJson(conf interface{}) error {
+	bytes, err := json.MarshalIndent(conf, "", "\t")
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintln(Stdout, string(bytes))
+	return nil
+}
+
+func (x *cmdGet) outputList(conf map[string]interface{}) error {
+	rootRequested := len(x.Positional.Keys) == 0
+	if rootRequested && len(conf) == 0 {
+		return fmt.Errorf("snap %q has no configuration", x.Positional.Snap)
+	}
+
+	w := tabWriter()
+	defer w.Flush()
+
+	fmt.Fprintf(w, "Key\tValue\n")
+	values := flattenConfig(conf, rootRequested)
+	for _, v := range values {
+		fmt.Fprintf(w, "%s\t%v\n", v.Path, v.Value)
+	}
+	return nil
+}
+
+var isTerminal = func() bool {
+	return terminal.IsTerminal(int(os.Stdin.Fd()))
+}
+
+func (x *cmdGet) outputDefault(conf map[string]interface{}, snapName string, confKeys []string) error {
```

Comment:
It took me a while to understand what is going on below (more like why than what). It would be great if we had a short blurb describing the desired behaviour.

Reply:
Thanks! I tried my best and added some extra description. However its still slightly complicated. I would love to simplify this further, I poked around a bit but failed so far to make it nicer. I may try again later again.